benchmark/util/text-decoder.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const common = require('../common.js');
+
+const bench = common.createBenchmark(main, {
+  encoding: ['utf-8', 'latin1', 'iso-8859-3'],
+  ignoreBOM: [0, 1],
+  len: [1, 64, 1024],
```

Comment:
```suggestion
  len: [256, 1024 * 16, 1024 * 512],
```

`1` is unlikely used in applications and even if, it would probably not be a bottleneck in that case. I suggest to try to use values that are in a bigger range and starting with values that are likely used more frequently for with end users. I hope the numbers above are nearer to real world usage.

Reply:
Thanks, @BridgeAR. You're the best. I'll update the benchmark result as well.

Reply:
```                                                                            confidence improvement accuracy (*)    (**)   (***)
util/text-decoder.js n=1000000 len=16384 ignoreBOM=0 encoding='iso-8859-3'                  8.90 %       ±9.65% ±12.86% ±16.76%
util/text-decoder.js n=1000000 len=16384 ignoreBOM=0 encoding='latin1'             ***     12.62 %       ±6.68%  ±8.89% ±11.58%
util/text-decoder.js n=1000000 len=16384 ignoreBOM=0 encoding='utf-8'              ***     17.13 %       ±7.98% ±10.61% ±13.81%
util/text-decoder.js n=1000000 len=16384 ignoreBOM=1 encoding='iso-8859-3'          **     15.27 %       ±9.90% ±13.19% ±17.19%
util/text-decoder.js n=1000000 len=16384 ignoreBOM=1 encoding='latin1'              **     11.16 %       ±8.25% ±10.98% ±14.29%
util/text-decoder.js n=1000000 len=16384 ignoreBOM=1 encoding='utf-8'                *     10.82 %       ±8.64% ±11.51% ±14.99%
util/text-decoder.js n=1000000 len=256 ignoreBOM=0 encoding='iso-8859-3'           ***     11.55 %       ±6.15%  ±8.19% ±10.66%
util/text-decoder.js n=1000000 len=256 ignoreBOM=0 encoding='latin1'                **     11.19 %       ±6.67%  ±8.90% ±11.64%
util/text-decoder.js n=1000000 len=256 ignoreBOM=0 encoding='utf-8'                ***     13.87 %       ±7.02%  ±9.34% ±12.17%
util/text-decoder.js n=1000000 len=256 ignoreBOM=1 encoding='iso-8859-3'           ***     16.50 %       ±3.39%  ±4.51%  ±5.87%
util/text-decoder.js n=1000000 len=256 ignoreBOM=1 encoding='latin1'               ***     15.83 %       ±6.26%  ±8.33% ±10.84%
util/text-decoder.js n=1000000 len=256 ignoreBOM=1 encoding='utf-8'                ***     13.83 %       ±6.36%  ±8.46% ±11.02%
util/text-decoder.js n=1000000 len=524288 ignoreBOM=0 encoding='iso-8859-3'         **      4.52 %       ±2.96%  ±3.93%  ±5.12%
util/text-decoder.js n=1000000 len=524288 ignoreBOM=0 encoding='latin1'              *      4.25 %       ±4.18%  ±5.57%  ±7.25%
util/text-decoder.js n=1000000 len=524288 ignoreBOM=0 encoding='utf-8'                      7.44 %       ±9.20% ±12.25% ±15.97%
util/text-decoder.js n=1000000 len=524288 ignoreBOM=1 encoding='iso-8859-3'         **      3.64 %       ±2.38%  ±3.16%  ±4.12%
util/text-decoder.js n=1000000 len=524288 ignoreBOM=1 encoding='latin1'                     0.69 %       ±2.98%  ±3.97%  ±5.17%
util/text-decoder.js n=1000000 len=524288 ignoreBOM=1 encoding='utf-8'                      1.90 %      ±10.00% ±13.31% ±17.33%

Be aware that when doing many comparisons the risk of a false-positive result increases.
In this case, there are 18 comparisons, you can thus expect the following amount of false-positive results:
  0.90 false positives, when considering a   5% risk acceptance (*, **, ***),
  0.18 false positives, when considering a   1% risk acceptance (**, ***),
  0.02 false positives, when considering a 0.1% risk acceptance (***)```